rust/bindings/rwonce.rs

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: GPL-2.0
+
+// Copyright (C) 2024 Google LLC.
+
+/// Read a value once.
+///
+/// # Safety
+///
+/// Same requirements as `READ_ONCE` from C.
+#[inline(always)]
+pub unsafe fn READ_ONCE<T>(ptr: *const T) -> T {
```

Comment:
In my patch: https://lore.kernel.org/lkml/20231025195339.1431894-1-boqun.feng@gmail.com/, I made it only work when `T: Copy`, so that we won't need to worry about cases where `T` has a `drop` function. I'm not sure whether the limitation provides any real world benefits, however, I think it's always good to start with some limitation and relax in the future if that seems too limited. Thoughts?

Reply:
Are these meant to have the same semantics as the C versions with the same name?

If not, I suggest we use a different name.

If so, on aarch64 with LTO enabled, READ_ONCE implies an acquire barrier: https://elixir.bootlin.com/linux/v6.9/source/arch/arm64/include/asm/rwonce.h#L36

Reply:
They're meant to have the same semantics as the C versions.

Reply:
I couldn't figure out how to replicate the CPU feature detection that C is doing, so I did something temporary instead. Please let me know if you know how to do the CPU feature detection properly.